Code/Framework/AzCore/AzCore/Instance/InstancePool.h

```diff
@@ -170,11 +170,25 @@ namespace AZ
         template<class T>
         AZStd::shared_ptr<InstancePool<T>> GetPool()
         {
-            const char* name = AZ::AzTypeInfo<T>::Name();
-            return GetPool<T>(AZ::Name(name));
+            return GetPool<T>(GetDefaultName<T>());
         }
 
     private:
+
+        template<class T>
+        AZ::Name GetDefaultName()
+        {
+            const char* name = AZ::AzTypeInfo<T>::Name();
+            auto moduleId = Environment::GetModuleId();
+
+            // the string length is 2x the pointer size (each byte takes 2 hexadecimal characters) + 2 bytes for the "0x"
+            const auto nameLength = sizeof(moduleId) * 2 + 2 + strlen(name);
+
+            AZStd::unique_ptr<char[]> fullname(new char[nameLength]);
+            azsnprintf(fullname.get(), nameLength, "%s%p", name, moduleId);
+            return AZ::Name(fullname.get());
```

Comment:
It is better to use an AZStd::fixed_string<256> here with the format function
```suggestion
            auto defaultName = AZStd::fixed_string<256>::format("%s%p", name, moduleId);
            return AZ::Name(fullname.get());
```